### ISSUE TITLE
[FEATURE] Migrate Secrets Manager Service to AWS SDK v2

### ISSUE THREAD

**Issue by aimethed** [awslabs/aws-athena-query-federation#2078]
Subtask of https://github.com/awslabs/aws-athena-query-federation/issues/1960

In order to update the federation sdk and connectors to AWS SDK v2, each service must be migrated. This issue is to migrate Secrets Manager to v2. This is done by removing the dependency on Secrets Manager v1, adding the dependency for v2, and then updating all uses in the codebase to v2.

See the AWS docs guide: https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/migration.html

**Comment by aimethed** [comment]
Addressed by PR linked. Work is finished, waiting on deployment changes before being merged into master.
